variables.tf

```diff
@@ -122,3 +116,20 @@ variable "ignore_changes_on_defaults" {
   type        = bool
   default     = false
 }
+
+variable "context_properties" {
+  description = "Specifies list of context properties used to create a Snowflake User name - this variable conflicts with `context_template`"
+  type        = list(string)
+  default     = ["stage", "name"]
```

Comment:
nb: thinking about ergonomics: if user wants to specify `template` then it has to nullify `properties` - fyi I don't have an answer, just thinking out loud 🤔

Reply:
Added logic, that nulls `context_properties` when proper context template is found (using `var.context_template_name` and `var.context_templates`)